packages/next/src/server/base-server.ts

```diff
@@ -2975,6 +2983,42 @@ export default abstract class Server<
       return null
     }
 
+    // If we're not in minimal mode and the cache entry that was returned was a
+    // app page fallback, then we need to kick off the dynamic shell generation.
+    if (
+      ssgCacheKey &&
+      !this.minimalMode &&
+      isRoutePPREnabled &&
+      this.nextConfig.experimental.pprFallbacks &&
+      cacheEntry.value?.kind === CachedRouteKind.APP_PAGE &&
+      cacheEntry.isFallback &&
+      !isOnDemandRevalidate
+    ) {
+      scheduleOnNextTick(async () => {
```

Comment:
Is this the right priority? this will run before the server can handle another request. it will also run before IO events which might resolve data on a concurrently rendering response. Feels like we want a lower priority background queue so actively rendering pages are prioritized over this?

Reply:
This is the same logic that's completed for existing background revalidates. I'd like to create a proper background queue for these types of operations, but want to split that into a future PR.